### PR TITLE
Add "Karmada + Flux" user guide

### PR DESCRIPTION
**Pull request by Poor12**
Add “Karmada + Flux” user guide to increase the use case of flux in multi-cluster application distribution.

**Special notes:**
/cc @stefanprodan

### CODE REVIEW

File: content/en/docs/use-cases/karmada.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+      kind: HelmRelease
+      name: podinfo
+  placement:
+    clusterAffinity:
+      clusterNames:
+        - member1
+        - member2
+```
+
+It means that the HelmRepository and HelmRelease created in Step 1 will be propagated to member1 and member2.
```
Comment (stefanprodan) [inline content/en/docs/use-cases/karmada.md:121]:
```suggestion
The above configuration is for propagating the Flux objects to member1 and member2 clusters.
```

File: content/en/docs/use-cases/karmada.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+3. Apply those manifests to the Karmada-apiserver:
+
+```sh
+kubectl apply -f ../helm/ --kubeconfig=/$HOME/.kube/karmada.config
+```
+
+The output is similar to:
+
+```
+helmrelease.helm.toolkit.fluxcd.io/podinfo created
```
Comment (stefanprodan) [inline content/en/docs/use-cases/karmada.md:132]: How come these Flux custom resources work on this cluster? Isn't the controller plane cluster getting filled with Helm release which shouldn't be reconcile here?

File: content/en/docs/use-cases/karmada.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+```
+
+### Flux Installation 
+
+In Karmada control plane, you only need to install Flux crds [here](https://github.com/fluxcd/flux2/blob/main/manifests/crds/kustomization.yaml) as objects delivered to member clusters.
+
+In member clusters, you can install Flux without storing its manifests in a Git repository in member clusters for testing purposes:
+
+```sh
+flux install
```
[the remark this answers is not in the document]
Comment (Poor12) [reply content/en/docs/use-cases/karmada.md:48]: Yeah, this should be ```flux install --kubeconfig=~/.kube/karmada.config --context=memberX```. I am missing. In Karmada control plane, we only apply flux crds but not need controllers to reconcile them. They are treated as resource template, not a resource instance. Based on work API [here](https://github.com/kubernetes-sigs/work-api), they will be encapsulated as a work object deliverd to member clusters and will be reconciled by controllers in member clusters finally.

[the remark this answers is not in the document]
Comment (stefanprodan) [reply content/en/docs/use-cases/karmada.md:48]: Ok so add the CRDs apply instructions as the first set:

```sh
 kubectl  apply -k github.com/fluxcd/flux2/manifests/crds?ref=main --kubeconfig=~/.kube/karmada.config
```

Explain what it does, then add the members commands after it.

Thanks!

File: content/en/docs/use-cases/karmada.md [hunk 4]
```diff
@@ -0,0 +1,198 @@
+---
+title: Karmada + Flux
+linkTitle: Karmada + Flux
+description: "How to use Karmada for distribution and management of multi-cluster helm charts."
```
Comment (stefanprodan) [inline content/en/docs/use-cases/karmada.md]:
```suggestion
description: "How to use Karmada for distribution and management of multi-cluster Flux Helm releases."
```

File: content/en/docs/use-cases/karmada.md [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+### Flux Installation 
+
+In Karmada control plane, you need to install Flux crds but do not need controllers to reconcile them. They are treated as resource templates, not specific resource instances. 
+Based on work API [here](https://github.com/kubernetes-sigs/work-api), they will be encapsulated as a work object deliverd to member clusters and reconciled by Flux controllers in member clusters finally.
+
+```sh
+kubectl apply -k github.com/fluxcd/flux2/manifests/crds?ref=main --kubeconfig ~/.kube/karmada.config
+```
+
+In member clusters, you can install Flux without storing its manifests in a Git repository in member clusters for testing purposes:
```
Comment (stefanprodan) [inline content/en/docs/use-cases/karmada.md:50]:
```suggestion
For testing purposes, we'll install Flux on member clusters without storing its manifests in a Git repository:
```

File: content/en/docs/use-cases/karmada.md [hunk 6]
```diff
@@ -0,0 +1,198 @@
+---
+title: Karmada + Flux
+linkTitle: Karmada + Flux
+description: "How to use Karmada for distribution and management of multi-cluster helm releases."
```
Comment (stefanprodan) [inline content/en/docs/use-cases/karmada.md:4]:
```suggestion
description: "How to use Karmada and Flux for distribution and management of multi-cluster Helm releases."
```

File: content/en/docs/use-cases/karmada.md [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+{{% alert color="info" title="Tip" %}}
+If you want to manage Helm releases across your fleet of clusters, Flux must be installed on each cluster.
+{{% /alert %}}
+
+## Helm release propagation
+
+If you want to propagate Helm releases for your apps to member clusters, you can refer to the guide below.
+
+1. Define a Flux `HelmRepository` manifest and a `HelmRelease` manifest in Karmada Control plane. They will serve as a resource template.
```
Comment (stefanprodan) [inline content/en/docs/use-cases/karmada.md:66]:
```suggestion
1. Define a Flux `HelmRepository` and a `HelmRelease` manifest in Karmada Control plane. They will serve as  resource templates. 
```

**Review by stefanprodan — COMMENTED** [review]
Please replace `helm` (which is the CLI) with `Helm` (which is the technology) in titles and text.

File: content/en/docs/use-cases/karmada.md [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+```sh
+kubectl apply -k github.com/fluxcd/flux2/manifests/crds?ref=main --kubeconfig ~/.kube/karmada.config
+```
+
+For testing purposes, we'll install Flux on member clusters without storing its manifests in a Git repository:
+
+```sh
+flux install --kubeconfig ~/.kube/members.config --context memberX
+```
```
Comment (stefanprodan) [inline content/en/docs/use-cases/karmada.md:54]:
```suggestion
```sh
flux install --kubeconfig ~/.kube/members.config --context member1
flux install --kubeconfig ~/.kube/members.config --context member2
```
So people can just copy/paste.
```

**Review by stefanprodan — COMMENTED** [review]
This looks Ok now. But we should add a disclaimer that this guide is not for doing GitOps. Since Flux is not installed on the management cluster, people have to manually apply the HelmReleases and the PropagationPolicies with kubectl, which goes against Flux & GitOps.

File: content/en/docs/use-cases/karmada.md [hunk 9]
```diff
@@ -0,0 +1,217 @@
+---
+title: Karmada + Flux
+linkTitle: Karmada + Flux
+description: "How to use Karmada and Flux for distribution and management of multi-cluster Flux Helm releases."
+weight: 20
+---
+
+{{% alert color="warning" title="Disclaimer" %}}
```
Comment (Poor12) [inline content/en/docs/use-cases/karmada.md:8]: How about adding a disclaimer here?

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @Poor12 🥇

File: content/en/docs/use-cases/karmada.md (hunk 9 above)
Comment (stefanprodan) [reply content/en/docs/use-cases/karmada.md:8]: Looks great, thanks!

**Pull request merged**